sdk/cosmos/azure-cosmos/test/test_container_properties_cache.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# The MIT License (MIT)
+# Copyright (c) Microsoft Corporation. All rights reserved.
+
+"""End-to-end test.
+"""
+
+import unittest
+import uuid
+
+import pytest
+
+import azure.cosmos.cosmos_client as cosmos_client
+import azure.cosmos.exceptions as exceptions
+import test_config
+from azure.cosmos.partition_key import PartitionKey
+
+
+@pytest.mark.cosmosEmulator
+class TestContainerPropertiesCache(unittest.TestCase):
+    """Python CRUD Tests.
+        """
+
+    configs = test_config.TestConfig
+    host = configs.host
+    masterKey = configs.masterKey
+    connectionPolicy = configs.connectionPolicy
+    last_headers = []
+    client: cosmos_client.CosmosClient = None
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
+                cls.host == '[YOUR_ENDPOINT_HERE]'):
+            raise Exception(
+                "You must specify your Azure Cosmos account values for "
+                "'masterKey' and 'host' at the top of this class to run the "
+                "tests.")
+        cls.client = cosmos_client.CosmosClient(cls.host, cls.masterKey)
+        cls.databaseForTest = cls.client.get_database_client(cls.configs.TEST_DATABASE_ID)
+
+    def test_container_properties_cache(self):
+        client = self.client
+        database_name = "Container Properties Cache Test DB " + str(uuid.uuid4())
+        created_db = client.create_database(database_name)
+        container_name = str(uuid.uuid4())
```

Comment:
there should be no need to re-declare the client here, or to create the DB if it will only be used for this test - let's instead initialize the database in the setUpClass and carry that forward through the test the way it is done in other tests